src/node_sqlite.cc

```diff
@@ -1860,6 +1895,112 @@ void DatabaseSync::LoadExtension(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  if (args[0]->IsNull()) {
+    // Clear the authorizer
+    sqlite3_set_authorizer(db->connection_, nullptr, nullptr);
+    db->object()->SetInternalField(kAuthorizerCallback, Null(isolate));
+    return;
+  }
+
+  if (!args[0]->IsFunction()) {
+    THROW_ERR_INVALID_ARG_TYPE(
+        isolate, "The \"callback\" argument must be a function or null.");
+    return;
+  }
+
+  Local<Function> fn = args[0].As<Function>();
+
+  db->object()->SetInternalField(kAuthorizerCallback, fn);
+
+  int r = sqlite3_set_authorizer(
+      db->connection_, DatabaseSync::AuthorizerCallback, db);
+
+  if (r != SQLITE_OK) {
+    CHECK_ERROR_OR_THROW(isolate, db, r, SQLITE_OK, void());
+  }
+}
+
+int DatabaseSync::AuthorizerCallback(void* user_data,
+                                     int action_code,
+                                     const char* param1,
+                                     const char* param2,
+                                     const char* param3,
+                                     const char* param4) {
+  DatabaseSync* db = static_cast<DatabaseSync*>(user_data);
+  Environment* env = db->env();
+  Isolate* isolate = env->isolate();
+  HandleScope handle_scope(isolate);
+  Local<Context> context = env->context();
+
+  Local<Value> cb =
+      db->object()->GetInternalField(kAuthorizerCallback).template As<Value>();
+
+  CHECK(cb->IsFunction());
+
+  Local<Function> callback = cb.As<Function>();
+  LocalVector<Value> js_argv(isolate);
+
+  // Convert SQLite authorizer parameters to JavaScript values
+  js_argv.emplace_back(Integer::New(isolate, action_code));
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param1).ToLocalChecked());
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param2).ToLocalChecked());
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param3).ToLocalChecked());
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param4).ToLocalChecked());
+
+  TryCatch try_catch(isolate);
+  MaybeLocal<Value> retval = callback->Call(
+      context, Undefined(isolate), js_argv.size(), js_argv.data());
+
+  if (try_catch.HasCaught()) {
+    db->StoreAuthorizerError(try_catch.Exception());
+    return SQLITE_DENY;
+  }
+
+  Local<Value> result;
+  if (!retval.ToLocal(&result) || result->IsUndefined() || result->IsNull()) {
+    // Missing return value
+    Local<Value> err = Exception::TypeError(
+        String::NewFromUtf8(
+            isolate,
+            "Authorizer callback must return an integer authorization code")
+            .ToLocalChecked());
+    db->StoreAuthorizerError(err);
+    return SQLITE_DENY;
+  }
+
+  if (!result->IsInt32()) {
+    Local<Value> err = Exception::TypeError(
+        String::NewFromUtf8(
+            isolate, "Authorizer callback return value must be an integer")
+            .ToLocalChecked());
```

Comment:
I think this is, in essence, the same error.

Reply:
The condition is: it has to be an integer and should be one of the valid options (SQLITE_OK, SQLITE_DENY, SQLITE_IGNORE). We can probably have this in a single if condition and simplify this validation.